### PR TITLE
Getting started info

### PR DESCRIPTION
**Pull request by mccalluc**
Hi! I was working on a tool to suggest concertina fingerings, and to provide tunes as input an ABC parser would be great, and your package looks like the best option in python. I'm just starting, but I think I may want to suggest a new feature at some point... but before we get there I wanted to document the steps a new developer should take, and check with you about the PR process that you'd like to follow.

Feel free to edit, or bounce back to me for changes, or close if it's not the direction you want.

To get myself oriented to the codebase, would it be ok if I tried to address one of the current issues? ...
- #40
- #38 
- #34

Would you like to assign me to one of those? Thanks.

### CODE REVIEW

**Review by zmoon — COMMENTED** [review]
Thanks @mccalluc , I added some initial comments.

---

> working on a tool to suggest concertina fingerings, and to provide tunes as input an ABC parser would be great, and your package looks like the best option in python.

Sounds cool, what type(s) of concertina?

And thanks! If you mostly just want to get at the notes, PyABC2 should work.

> I'm just starting, but I think I may want to suggest a new feature at some point... 

That would certainly be welcome!


> To get myself oriented to the codebase, would it be ok if I tried to address one of the current issues? ...
> https://github.com/zmoon/PyABC2/issues/40
> https://github.com/zmoon/PyABC2/issues/38
> https://github.com/zmoon/PyABC2/issues/34
> Would you like to assign me to one of those? Thanks.

Will do

**Comment by mccalluc** [comment]
Thank you for the feedback! I'm going to mark this as draft, fix all the typos, then move the whole section to a new document, as you suggest.

> Sounds cool, what type(s) of concertina?

Anglo, so there are different notes on the push and the pull. There are some other projects out there that already do this in different ways, but seems like a tool with a good command line interface would still be useful:

- [Anglo Concertina Fingering Generator](https://jvandonsel.github.io/fingering/fingering.html): Web page which takes ABC notation, and returns ABC notation, and renders it using [abcjs](https://www.abcjs.net/)
- [concertina-pbqp](https://github.com/resistor/concertina-pbqp): C++; Models it as an NP-hard problem, but uses a solver library for an approximate solution. 
- [juliamullen/concertina](https://github.com/juliamullen/concertina): Python classes. Melody needs to be written into source code.

File: README.md [hunk 1]
```diff
@@ -7,6 +7,43 @@
...
+Then set up a virtual env using venv or conda. For example:
+```
+python3 -m venv env
+source env/bin/activate
+```
+
+Then install dependencies and pre-commit hooks:
+```
+pip install flit
+flit install
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply README.md:36]: Good thought: I'll spell it out so it's more clear what the option does.
```suggestion
flit install --symlink
```

File: pyproject.toml [hunk 2]
```diff
@@ -33,6 +33,7 @@ test = [
 dev = [
     "ipython",
     "jupyterlab",
+    "pre-commit",
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply pyproject.toml:36]: Sure, I'll link to the docs instead.
```suggestion
```

**Comment by zmoon** [comment]
I play C/G Anglo a bit, interested to see what you come up with. That PBQP approach is interesting.

**Review by zmoon — APPROVED** [review]
Thanks a lot.

I plan to create Sphinx documentation in the not-too-distance future, and creating the `docs` folder is an important step towards this, so I appreciate it.

**Pull request merged**